packages/thirdweb/src/extensions/prebuilts/deploy-published.ts

```diff
@@ -0,0 +1,146 @@
+import type { AbiConstructor } from "abitype";
+import type { Chain } from "../../chains/types.js";
+import type { ThirdwebClient } from "../../client/client.js";
+import { getContract } from "../../contract/contract.js";
+import { deployViaAutoFactory } from "../../contract/deployment/deploy-via-autofactory.js";
+import { deployContract } from "../../contract/deployment/deploy-with-abi.js";
+import { fetchPublishedContractMetadata } from "../../contract/deployment/publisher.js";
+import { getOrDeployInfraForPublishedContract } from "../../contract/deployment/utils/bootstrap.js";
+import { sendAndConfirmTransaction } from "../../transaction/actions/send-and-confirm-transaction.js";
+import { simulateTransaction } from "../../transaction/actions/simulate.js";
+import { prepareContractCall } from "../../transaction/prepare-contract-call.js";
+import { resolveMethod } from "../../transaction/resolve-method.js";
+import type { Account } from "../../wallets/interfaces/wallet.js";
+
+export type DeployPublishedContractOptions = {
+  client: ThirdwebClient;
+  chain: Chain;
+  account: Account;
+  contractId: string;
+  contractParams: unknown[];
+  publisher?: string;
+  version?: string;
+  implementationConstructorParams?: unknown[];
+};
+
+/**
+ * Deploy an instance of a published contract on a given chain
+ * @param options - the deploy options
+ * @returns a promise that resolves to the deployed contract address
+ * @example
+ * ```
+ * import { deployPublishedContract } from "thirdweb/deploys";
+ *
+ * const address = await deployedPublishedContract({
+ *   client,
+ *   chain,
+ *   account,
+ *   contractId: "MyPublishedContract",
+ *   contractParams: [...],
+ *   publisher: "0x...",
+ * });
+ * ```
+ * @extension DEPLOY
+ */
+export async function deployPublishedContract(
+  options: DeployPublishedContractOptions,
+): Promise<string> {
+  const {
+    client,
+    account,
+    chain,
+    contractId,
+    contractParams,
+    publisher,
+    version,
+    implementationConstructorParams,
+  } = options;
+  const { compilerMetadata, extendedMetadata } =
+    await fetchPublishedContractMetadata({
+      client,
+      contractId,
+      publisher,
+      version,
+    });
+  if (extendedMetadata?.deployType === "standard") {
```

Comment:
it _feels_ to me like a switch case on the deployType would be more readable here?